common/djangoapps/student/tests/test_reset_password.py

```diff
@@ -252,6 +252,23 @@ def test_reset_password_good_token(self):
         self.user = User.objects.get(pk=self.user.pk)
         self.assertTrue(self.user.is_active)
 
+    def test_password_rest_fail(self):
+        """Tests that if we providing mismatching passwords user is not marked as active"""
```

Comment:
Also, note that the user is marked as active as soon as they click on their password-reset link in the email.  So even before they get to the page where they need to enter their new password.  
Given this, this particular testing of the `is_active` field is somewhat misleading.